src/index.ts

```diff
@@ -231,31 +229,43 @@ export const handleRotateKey = async (ctx: Context, _request?: Request) => {
 
 const handleClearKey = async (ctx: Context, _request?: Request) => {
 	ctx.metrics.keyClearTotal.inc();
-
-	const now = Date.now();
+	const now = new Date();
 
 	const keys = await ctx.bucket.ISSUANCE_KEYS.list({ shouldUseCache: false });
 
+	if (keys.objects.length === 0) {
+		return new Response('No keys to clear', { status: 201 });
+	}
+
+	// Find the latest key based on the upload time
 	let latestKey = keys.objects[0];
-	const toDelete: Set<string> = new Set();
+	let mostRecentKeyUploadTime = new Date(latestKey.uploaded); // Assuming this is the last rotation
 
 	for (const key of keys.objects) {
-		const keyUploadTime = new Date(key.uploaded).getTime();
-		const keyExpirationTime = keyUploadTime + KEY_LIFESPAN;
-
-		if (keyExpirationTime < now) {
-			toDelete.add(key.key);
+		const keyUploadTime = new Date(key.uploaded);
+		if (keyUploadTime.getTime() > mostRecentKeyUploadTime.getTime()) {
+			mostRecentKeyUploadTime = keyUploadTime;
+			latestKey = key;
```

Comment:
why having two variables while one is simply `new Date(latestKey.uploaded)`?

Reply:
Could we use `Array,reduce()` here?
```
  let latestKey = keys.objects.reduce((latest, current) => {
        return new Date(current.uploaded) > new Date(latest.uploaded) ? current : latest;
    });
```

Reply:
reduce with a ternary operator seems harder to read than a for loop